From d8d29e9d68bec491aee6f1e871a9c490aa2481b1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marko=20Luk=C5=A1a?= <marko.luksa@gmail.com>
Date: Wed, 2 Mar 2022 19:16:39 +0100
Subject: OSSM-1074 Fix rendering of injectedAnnotations in Helm charts (#473)

If an injected annotation value contained a double-quote character, the rendered istiod-injector-configmap YAML would be invalid. Unmarshalling the inject config would fail with: update error: failed to unmarshal injection template: error converting YAML to JSON: yaml: line 10: did not find expected key
---
 .../istio-discovery/templates/istiod-injector-configmap.yaml    | 2 +-
 .../istiod-remote/templates/istiod-injector-configmap.yaml      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/manifests/charts/istio-control/istio-discovery/templates/istiod-injector-configmap.yaml b/manifests/charts/istio-control/istio-discovery/templates/istiod-injector-configmap.yaml
index b6b1fa8e80..1378a1f629 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/istiod-injector-configmap.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/istiod-injector-configmap.yaml
@@ -36,7 +36,7 @@ data:
 {{ toYaml .Values.sidecarInjectorWebhook.neverInjectSelector | trim | indent 6 }}
     injectedAnnotations:
       {{- range $key, $val := .Values.sidecarInjectorWebhook.injectedAnnotations }}
-      "{{ $key }}": "{{ $val }}"
+      "{{ $key }}": {{ $val | quote }}
       {{- end }}
     {{- /* If someone ends up with this new template, but an older Istiod image, they will attempt to render this template
          which will fail with "Pod injection failed: template: inject:1: function "Istio_1_9_Required_Template_And_Version_Mismatched" not defined".
diff --git a/manifests/charts/istiod-remote/templates/istiod-injector-configmap.yaml b/manifests/charts/istiod-remote/templates/istiod-injector-configmap.yaml
index b6b1fa8e80..1378a1f629 100644
--- a/manifests/charts/istiod-remote/templates/istiod-injector-configmap.yaml
+++ b/manifests/charts/istiod-remote/templates/istiod-injector-configmap.yaml
@@ -36,7 +36,7 @@ data:
 {{ toYaml .Values.sidecarInjectorWebhook.neverInjectSelector | trim | indent 6 }}
     injectedAnnotations:
       {{- range $key, $val := .Values.sidecarInjectorWebhook.injectedAnnotations }}
-      "{{ $key }}": "{{ $val }}"
+      "{{ $key }}": {{ $val | quote }}
       {{- end }}
     {{- /* If someone ends up with this new template, but an older Istiod image, they will attempt to render this template
          which will fail with "Pod injection failed: template: inject:1: function "Istio_1_9_Required_Template_And_Version_Mismatched" not defined".
-- 
2.31.1

